backend/pkg/api/handlers/internal.go

```diff
@@ -966,24 +967,26 @@ func (h *HandlerService) InternalGetValidatorDashboardSummaryChart(w http.Respon
 	groupIds := v.checkGroupIdList(q.Get("group_ids"))
 	efficiencyType := checkEnum[enums.VDBSummaryChartEfficiencyType](&v, q.Get("efficiency_type"), "efficiency_type")
 	aggregation := checkEnum[enums.ChartAggregation](&v, q.Get("aggregation"), "aggregation")
-	maxAge := getMaxChartAge(aggregation, premiumPerks.ChartHistorySeconds)
-	now := uint64(time.Now().Unix())
-	if now < maxAge {
-		maxAge = now
+	maxAge := getMaxChartAge(aggregation, premiumPerks.ChartHistorySeconds) // can be max int for unlimited, always check for underflows
+	if maxAge == 0 {
+		returnConflict(w, fmt.Errorf("requested aggregation is not available for dashboard owner's premium subscription"))
+		return
 	}
-	minAllowedTs := now - maxAge
-	afterTs, beforeTs := v.checkTimestamps(q.Get("after_ts"), q.Get("before_ts"), minAllowedTs)
-	if v.hasErrors() {
-		handleErr(w, v)
+	latestExportedTs, err := h.dai.GetLatestExportedChartTs(ctx, aggregation)
+	if err != nil {
+		handleErr(w, err)
 		return
 	}
-	if maxAge == 0 {
-		returnConflict(w, fmt.Errorf("requested aggregation is not available for dashboard owner's premium subscription"))
+	minAllowedTs := latestExportedTs - min(maxAge, latestExportedTs)                                      // min to prevent underflow
+	secondsPerEpoch := uint64(12 * 32)                                                                    // TODO: fetch dashboards chain id and use correct value for network once available
+	maxAllowedInterval := chartDatapointLimit*uint64(aggregation.Duration(secondsPerEpoch).Seconds()) - 1 // -1 to make sure we don't go over the limit
+	afterTs, beforeTs := v.checkTimestamps(q.Get("after_ts"), q.Get("before_ts"), latestExportedTs, minAllowedTs, maxAllowedInterval)
```

Comment:
Maybe it's possible to make all this logic for timestamps easier

Reply:
Prob not, but could try to plan for minimizing code duplication with other charts later. I'd probably wrap it all up in a general `chartTimestampHandler`<sup>TM</sup> method (or maybe 2 if we'll have charts without aggregation), and incorporate `checkTimestamps` into that function as I doubt that it will ever be used in another context.

Reply:
we'll do that once it happens :)